### PR TITLE
Fixing Tokernizer plugin so that spark-plugins are loaded on all the distros

### PR DESCRIPTION
**Pull request by CuriousVini**
Issue: https://issues.cask.co/browse/HYDRATOR-1116
Build: http://builds.cask.co/browse/HYP-BAD172-2

Tested it on CDH 5.4 cluster.

### CODE REVIEW

**Comment by albertshau** [comment]
How does this fix the problem?

**Comment by albertshau** [comment]
Oh it will deploy properly, but this plugin will still be broken right?

**Comment by CuriousVini** [comment]
@albertshau The problem was because `RegexTokenizer` was an instance variable of `Tokenizer` class, the spark-plugins jar was not getting loaded  So when `Tokenizer` is loaded  `RegexTokenizer` class was also getting loaded. We changed it to use `Object` and type cast it when needed so the spark-plugins jar is loaded and other tests are not affected on CDH 5.4 and HDP 2.2

**Comment by anwar6953** [comment]
"Oh it will deploy properly, but this plugin will still be broken right?"
Yep. It's just to unblock the other spark plugins which normally work on certains distros (CDH 5.2, CDH 2.4, I believe).

**Comment by albertshau** [comment]
ok I see. lgtm

File: spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/Tokenizer.java [hunk 1]
```diff
@@ -47,7 +47,8 @@
   public static final String PLUGIN_NAME = "Tokenizer";
   private Config config;
   private Schema outputSchema;
-  private RegexTokenizer tokenizer;
+  // Work around because of HYDRATOR-1116. Remove this change once its fixed
+  private Object tokenizer;
```
Comment (chtyim) [inline spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/Tokenizer.java]: Seems like this doesn't need to be a field at all. Just new it in the RDD `transform` function.

**Comment by CuriousVini** [comment]
@chtyim Addressed comments and tested on cdh 5.4 cluster. Please take another pass when you get a chance.

**Comment by CuriousVini** [comment]
Build is green

**Pull request merged**